Use Macro for __FILE__ & __LINE__ when possible

Baoulettes opened the pull request:
This PR has for goal to replace all hard coded line number and file path to a macro
Should be made across the whole ./soh/ folder 
It remove if(0){} and if {1} {} as requested.
To review it I suggest github desktop, I doubt a normal internet browser can handle it^^

Baoulettes committed:
First batch some overlay

Baoulettes committed:
Almost all overlay

Baoulettes committed:
effect & gamestate

Baoulettes committed:
kaleido stuffs

Baoulettes committed:
more overlay

Baoulettes committed:
more left over from code folder

Baoulettes committed:
remaining hardcoded line and file

Baoulettes committed:
Open & Close _DISP __FILE__ & __LINE__ clean up

Baoulettes committed:
Some if (1) {} remove

Baoulettes committed:
LOG_xxxx __FILE__ , __LINE__ cleaned

Emill commented:
The only issue with this is that it will be much harder to merge upstream code later on. Does the benefits of this PR really outweigh that?

Baoulettes committed:
ASSERT macro __FILE__ __LINE__

Baoulettes committed:
mtx without line/file in functions

Baoulettes committed:
" if (1) {} " & "if (0) {}" and tab/white place

Baoulettes commented:
updated draft with many more things I leave the message here in case there is something to change or correct 

I have no clue if there is remaining things I tried to search with ``grep -ir`` over all soh folder I do not see more things.
If anyone check the pr and spot and tab or indentation that is not correct (that I myself modified) feel free to point me, I tried to check everything.

I do not remove the draft mode I do not want Jenkins to compile it before I have green light :)

Baoulettes committed:
Merge branch 'HarbourMasters:develop' into openclose_disp_macro

Baoulettes committed:
LogUtils as macro

Baoulettes committed:
GameState_, GameAlloc_, SystemArena_ & ZeldaArena_

Baoulettes committed:
Revert "GameState_, GameAlloc_, SystemArena_ & ZeldaArena_"

This reverts commit 0d85caaf7e342648c01a15fe21e93637352dc596.

Baoulettes committed:
Like last commit but as macro

Kenix3 commented on soh/include/functions.h:
```diff
@@ -1763,8 +1763,8 @@ void Matrix_RotateZYX(s16 x, s16 y, s16 z, u8 mode);
-Mtx* Matrix_ToMtx(Mtx* dest, char* file, s32 line);
-Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx, char* file, s32 line);
+Mtx* Matrix_ToMtx(Mtx* dest);
+Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx);
```
We should never be removing the file and line arguments from a function call. This should still take a file and line macro with a macro function call. I did notice these matrix functionsake other calls. For keeping the same semantics, you'd have the matrix functions use the actual function call to other matrix calls rather than the macro that obfuscates file and line.

Kenix3 commented on soh/src/code/sys_matrix.c:
```diff
@@ -552,15 +552,15 @@ Mtx* Matrix_MtxFToMtx(MtxF* src, Mtx* dest) {
-    FrameInterpolation_RecordMatrixToMtx(dest, file, line);
-    guMtxF2L(Matrix_CheckFloats(sCurrentMatrix, file, line), dest);
+Mtx* Matrix_ToMtx(Mtx* dest) {
+    FrameInterpolation_RecordMatrixToMtx(dest, __FILE__, __LINE__);
```
For example, you'd just continue to pass in the file and line function arguments to this frame interpolation function here.

Kenix3 commented on soh/src/code/sys_matrix.c:
```diff
@@ -552,15 +552,15 @@ Mtx* Matrix_MtxFToMtx(MtxF* src, Mtx* dest) {
 
-Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx, char* file, s32 line) {
-    return Matrix_ToMtx(Graph_Alloc(gfxCtx, sizeof(Mtx)), file, line);
+Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx) {
```
```suggestion
Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx, char* file, s32 line) {
```

Kenix3 commented on soh/src/code/sys_matrix.c:
```diff
@@ -552,15 +552,15 @@ Mtx* Matrix_MtxFToMtx(MtxF* src, Mtx* dest) {
-Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx, char* file, s32 line) {
-    return Matrix_ToMtx(Graph_Alloc(gfxCtx, sizeof(Mtx)), file, line);
+Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx) {
+    return Matrix_ToMtx(Graph_Alloc(gfxCtx, sizeof(Mtx)));
```
```suggestion
    return Matrix_ToMtx(Graph_Alloc(gfxCtx, sizeof(Mtx)), file, line);
```

Here's an actual example. This will preserve the line number in say actor code or engine code that actually called this chain of matrix functions. It will also keep the same semantics to what the original code did.

Baoulettes committed:
Fix matrix not using macros

Baoulettes committed:
Merge branch 'HarbourMasters:develop' into openclose_disp_macro

Kenix3 commented on soh/src/code/sys_matrix.c:
```diff
@@ -552,15 +552,15 @@ Mtx* Matrix_MtxFToMtx(MtxF* src, Mtx* dest) {
+Mtx* Matrix_ToMtx(Mtx* dest, char* file, s32 line) {
     FrameInterpolation_RecordMatrixToMtx(dest, __FILE__, __LINE__);
-    guMtxF2L(Matrix_CheckFloats(sCurrentMatrix), dest);
+    guMtxF2L(MATRIX_CHECKFLOATS(sCurrentMatrix), dest);
```
You would not call the macro version of the function call here. Just pass through file and line function arguments.

Baoulettes committed:
use function not macro

Baoulettes committed:
DebugArena_* functions
GameAlloc_MallocDebug
BgCheck_PosErrorCheck as macros
removed issues with ; in macro file

Baoulettes committed:
Merge branch 'HarbourMasters:develop' into openclose_disp_macro
